azure-devops/azext_devops/dev/common/github_credential_manager.py

```diff
@@ -0,0 +1,79 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import base64
+import requests
+from knack.prompting import prompt, prompt_pass
+from knack.log import get_logger
+from knack.util import CLIError
+from azext_devops.dev.common.utils import randomword, singleton
+
+logger = get_logger(__name__)
+
+
+@singleton
+class GithubCredentialManager():
+    """ GithubCredentialManager
+    """
+    def __init__(self):
+        self.username = None
+        self.password = None
+        self.token = None
+
+    def create_token(self, note=None):
+        self.username = prompt(msg='Enter your GitHub username (leave blank for using already generated PAT): ')
```

Comment:
when asking for PAT, should we also give expectation on the scopes required? users might not be very happy to give PAT will all scopes

Reply:
Good point!
Though I was printing the warning message with the scopes and name of PAT created.
Moving it to before asking the credentials.